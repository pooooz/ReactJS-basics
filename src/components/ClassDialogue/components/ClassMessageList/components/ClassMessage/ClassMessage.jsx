import React, {Component} from "react";
import styles from './ClassMessage.module.scss'

export class ClassMessage extends Component {
  render() {
    return (
      <div className={styles.message}>
        <p className={styles.glitch} data-text={this.props.message}>{this.props.message}</p>
        <p className={styles.glitch} data-text={'Author: <' + this.props.author + '>'}>Author: &lt;{this.props.author}&gt;</p>
      </div>
    );
  }
}