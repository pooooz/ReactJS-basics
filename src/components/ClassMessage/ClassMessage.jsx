import React, {Component} from "react";
import styles from './ClassMessage.module.scss'

export class ClassMessage extends Component {
  render() {
    return <h1 className={styles.glitch} data-text={this.props.message}>{this.props.message}</h1>
  }
}