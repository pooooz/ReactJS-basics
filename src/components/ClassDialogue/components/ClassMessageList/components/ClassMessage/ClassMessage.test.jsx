import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { ClassMessage } from './ClassMessage';

describe('ClassMessage', () => {
  it('Render component', () => {
    render(<ClassMessage />);
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(<ClassMessage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render component with author', () => {
    render(<ClassMessage author="Me" />);
    expect(screen.getByText(/Author: <\w*>/)).toBeInTheDocument();
  });
});
