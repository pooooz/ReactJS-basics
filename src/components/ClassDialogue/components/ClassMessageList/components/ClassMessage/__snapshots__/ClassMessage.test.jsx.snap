// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassMessage Render with snapshot 1`] = `
<DocumentFragment>
  <div>
    <p />
    <p
      data-text="Author: <undefined>"
    >
      Author: &lt;&gt;
    </p>
  </div>
</DocumentFragment>
`;
